src/Admin/AdminConsole/Controllers/ProvidersController.cs

```diff
@@ -290,25 +291,43 @@ public async Task<IActionResult> Edit(Guid id, ProviderEditModel model)
 
         var providerPlans = await _providerPlanRepository.GetByProviderId(id);
 
-        if (providerPlans.Count == 0)
+        if (provider.IsBillable() && providerPlans.Count == 0)
         {
-            var newProviderPlans = new List<ProviderPlan>
-            {
-                new () { ProviderId = provider.Id, PlanType = PlanType.TeamsMonthly, SeatMinimum = model.TeamsMonthlySeatMinimum, PurchasedSeats = 0, AllocatedSeats = 0 },
-                new () { ProviderId = provider.Id, PlanType = PlanType.EnterpriseMonthly, SeatMinimum = model.EnterpriseMonthlySeatMinimum, PurchasedSeats = 0, AllocatedSeats = 0 }
-            };
-
-            foreach (var newProviderPlan in newProviderPlans)
-            {
-                await _providerPlanRepository.CreateAsync(newProviderPlan);
-            }
+            return Conflict("Provider must have at least one plan to be billable.");
         }
-        else
-        {
-            await _providerBillingService.UpdateSeatMinimums(
-                provider,
-                model.EnterpriseMonthlySeatMinimum,
-                model.TeamsMonthlySeatMinimum);
+
+        switch (provider.Type)
+        {
+            case ProviderType.Msp:
+                var updateMspSeatMinimumsCommand = new UpdateProviderSeatMinimumsCommand(
+                    provider.Id,
+                    new Dictionary<PlanType, int>
+                    {
+                        { PlanType.TeamsMonthly, model.TeamsMonthlySeatMinimum },
+                        { PlanType.EnterpriseMonthly, model.EnterpriseMonthlySeatMinimum }
+                    });
+                await _providerBillingService.UpdateSeatMinimums(updateMspSeatMinimumsCommand);
+                break;
+            case ProviderType.MultiOrganizationEnterprise:
+                {
+                    var existingMoePlan = providerPlans.Single();
+
+                    // 1. Change the plan and take over any old values.
+                    var changeMoePlanCommand = new ChangeProviderPlanCommand(
```

Comment:
❓ I really like these implementations, but one thing I'm concerned with here is the number of prorations we're going to be generating on Provider subscriptions with these changes back to back. What do you think of the effort to combine them into a singular subscription update? The resulting method (combined from `ChangePlan` and `UpdateSeatMinimum` would just need to ensure it ignores plan changes for MSPs which aren't possible. Obviously I don't want to impede this PR, but with these invoices going directly to a customer, they'll be sensitive to the confusing number of line items that might show here. Can you run one of these updates locally to show what those prorations might look like?

Reply:
We discussed this in Slack. Let's leave it as is and see how the Product team feels about it during internal testing. I think it should be fine.